src/lib.rs

```diff
@@ -69,19 +68,21 @@ impl Sharks {
     ///
     /// Example:
     /// ```
-    /// # use sharks::Sharks;
+    /// # use sharks::{ Sharks, Share };
     /// # let sharks = Sharks(3);
-    /// # let mut shares = sharks.dealer(&[1]).take(3).collect();
-    /// // Revover original secret from shares
+    /// # let mut shares: Vec<Share> = sharks.dealer(&[1]).take(3).collect();
+    /// // Recover original secret from shares
     /// let mut secret = sharks.recover(&shares);
     /// // Secret correctly recovered
     /// assert!(secret.is_ok());
-    /// // Remove shares for demonstrastion purposes
+    /// // Remove shares for demonstration purposes
     /// shares.clear();
     /// secret = sharks.recover(&shares);
     /// // Not enough shares to recover secret
     /// assert!(secret.is_err());
-    pub fn recover(&self, shares: &HashMap<GF256, Vec<GF256>>) -> Result<Vec<u8>, &str> {
+    pub fn recover(&self, shares: &[Share]) -> Result<Vec<u8>, &str> {
```

Comment:
Here I used `HashMap` to enforce all shares to be different. If for example, the minimum threshold is 3, but we have two duplicated shares, the password is still irrecoverable, even if the length of the list was 3.

We should discuss if changing to a slice is beneficial, or we should go back to using the `HashMap`, because right now, the check `shares.len() < self.0` is somewhat useless.

Reply:
Oh yeah, I see it, that's something I didn't catch!

One naive way to do this would be to redo the length check manually(i could easily write a small algorithm that checks for X duplicates, it isn't really hard), but that might incur a small performance drop compared to the HashMap.

Another way would be to make a HashSet from the `x` part and check the length of that. I believe that would be simpler and more effective that even the HashMap.

One thing I don't really like about directly sending an HashMap to the public API is simply about the complexity for the user. It is a bit harder to use for the user than a simple slice, and also you might want tit to be easy to use even for people who don't understand the math behind it. If you accept a slice, it abstracts to the user of the library the coordinate aspect of the share.

In the end that's your call, but I do believe the best would be to replace the `shares.len() < self.0` by a similar check but on a HashSet. I can commit the change, it's not a problem!

Reply:
Great! I agree with you, let's put the `x` values in a `HashSet` and compare its length 👍 

Please, also add a unit test to check it works. Many thanks!

Reply:
What I'm suggesting is this(works as expected with my manual tests):
`
pub fn recover(&self, shares: &[Share]) -> Result<Vec<u8>, &str> {
    let shares_x: HashSet<u8> = HashSet::from(shares.iter().map(|s| s.x.0).collect());

    if shares_x.len() < self.0 as usize {
        Err("Not enough shares to recover original secret")
    } else {
        Ok(math::interpolate(shares))
    }
}
`

Reply:
Done and committed!

Reply:
Perfect! I see you didn't use `from` in the end, that would have added overhead :)

I am going to approve your PR :heart:

Reply:
Clippy warned me this was useless! Didn't know I could simply collect into a HashSet!